syncapi/sync/requestpool.go

```diff
@@ -328,7 +328,9 @@ func (rp *RequestPool) appendAccountData(
 	// data keys were set between two message. This isn't a huge issue since the
 	// duplicate data doesn't represent a huge quantity of data, but an optimisation
 	// here would be making sure each data is sent only once to the client.
-	if req.since == nil {
+	// TODO: We used to have req.since == nil here, is there any case where req.since
+	// can actually be nil?
+	if req.since.PDUPosition() == 0 && req.since.EDUPosition() == 0 {
```

Comment:
```suggestion
	if req.since == nil || (req.since.PDUPosition() == 0 && req.since.EDUPosition() == 0) {
```
Since we can't be sure `req.since != nil` (makes that TODO obsolete)

Reply:
Done in ff787cc.